service/interfaces.go

```diff
@@ -23,3 +26,15 @@ type HealthChecker interface {
 	Stop()
 	AddCheck(name string, checker healthcheck.Checker) (err error)
 }
+
+// CantabularClient wraps the CantabularClient handler interface, adding the Checker method
+type CantabularClient interface {
+	handler.CantabularClient
```

Comment:
I don't think it's good to couple packages and interfaces like this, it's better if each package is responsible for defining it's own interfaces.
It could soon get messy if different packages started to need different but overlapping parts of the main interface.

Reply:
I was assuming that the service package interface will always be a superset of the handler interface (the handler interface defines the API related stuff, and the service is a wrapper that adds the Checker func), but I can decouple it